File: src/00/sigma_core/src/ontology_gen/00/normalize_angle.rs
#![allow(unused_imports)]

pub fn normalize_angle(angle: f64) -> f64 {
    let tau = 2.0 * std::f64::consts::PI;
    let mut a = angle % tau;
    if a < 0.0 {
        a += tau;
    }
    a / tau
}

File: src/_as/00/normalize_angle.ts
// SSoT: src/ontology/math/normalize_angle.md

@inline
export function normalize_angle(angle: f64): f64 {
const tau: f64 = 2.0 * Math.PI;
let a: f64 = angle % tau;
if (a < 0.0) a += tau;
return a / tau;
}
